### PR TITLE
chore: snapshot performance tracking improvements

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

We're still seeing some high loading times for snapshots

## Changes

Track some more things and try to fix some bugs (documented in comments on relevant lines)

## How did you test this code?

Not necessary

### CODE REVIEW

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 1]
```diff
@@ -93,6 +93,7 @@ interface RecordingViewedProps {
     performance_events_load_time: number // How long it took to load all performance events
     first_paint_load_time: number // How long it took to first contentful paint (time it takes for user to see first frame)
     duration: number // How long is the total recording (milliseconds)
+    recording_id: string // End timestamp of the session
```
Comment (daibhin) [inline frontend/src/lib/utils/eventUsageLogic.ts:96]:
```suggestion
    recording_id: string // Id of the session
```

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 2]
```diff
@@ -846,6 +847,7 @@ export const eventUsageLogic = kea<eventUsageLogicType>([
                 events_load_time: durations.events,
                 first_paint_load_time: durations.firstPaint,
                 duration: eventIndex.getDuration(),
+                recording_id: playerData.sessionRecordingId,
```
Comment (daibhin) [inline frontend/src/lib/utils/eventUsageLogic.ts:850]: Track the id of the specific recording (might be necessary to debug)

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts [hunk 3]
```diff
@@ -537,6 +540,7 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
 
                         posthog.capture('recording_snapshot_loaded', {
                             source: source.source,
+                            duration: Math.round(performance.now() - snapshotLoadingStartTime),
```
Comment (daibhin) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:543]: Add timings for individual sources. Trying to understand which sources are the slowest to load.

Note: `snapshotLoadingStartTime` resets every time `loadRecordingSnapshots` is called, but `cache.snapshotsStartTime` does not as it measures the time the first snapshot source loading was started

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts [hunk 4]
```diff
@@ -830,6 +837,9 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
             },
         ],
     }),
+    afterMount(({ cache }) => {
```
Comment (daibhin) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:840]: Possibly some browsers with stale `cache` values that are being populated on load (e.g. values not removed in the `beforeUnmount` somehow). Reset all timing metrics on mount

File: frontend/__snapshots__/replay-listings--recordings-play-lists--dark.png
Comment (daibhin) [inline frontend/__snapshots__/replay-listings--recordings-play-lists--dark.png:1]: Don't know why they've decided to change on this PR but snapshots are valid. We increased the contrast or icons a while ago

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts (hunk 4 above)
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:840]: 🧠

**Review by pauldambra — APPROVED** [review]
🚢

**Pull request merged**